src/shared_modules/content_manager/src/components/CtiDownloader.hpp

```diff
@@ -85,22 +85,50 @@ class CtiDownloader : public AbstractHandler<std::shared_ptr<UpdaterContext>>
      */
     CtiBaseParameters getCtiBaseParameters(const std::string& ctiURL)
     {
-        CtiBaseParameters parameters;
+        nlohmann::json rawMetadata;
 
         // Routine that stores the necessary parameters.
-        const auto onSuccess {[&parameters](const std::string& response)
+        const auto onSuccess {[&rawMetadata](const std::string& response)
                               {
-                                  const auto responseData = nlohmann::json::parse(response).at("data");
+                                  logDebug2(WM_CONTENTUPDATER, "CTI raw metadata: '%s'", response.c_str());
+
+                                  if (!nlohmann::json::accept(response))
+                                  {
+                                      throw std::runtime_error {"Invalid CTI metadata format"};
+                                  }
 
-                                  parameters.lastOffset = responseData.at("last_offset").get<int>();
-                                  parameters.lastSnapshotLink =
-                                      responseData.at("last_snapshot_link").get<std::string>();
-                                  parameters.lastSnapshotOffset = responseData.at("last_snapshot_offset").get<int>();
+                                  rawMetadata = nlohmann::json::parse(response).at("data");
```

Comment:
Maybe we could check that the response contains the "data" field and if not, throw a std::runtime_error {"Invalid CTI metadata format"} exception too.